lib/rails-jquery-autocomplete/autocomplete.rb

```diff
@@ -96,12 +96,12 @@ def get_object(model_sym)
     # Can be overriden to show whatever you like
     # Hash also includes a key/value pair for each method in extra_data
     #
-    def json_for_autocomplete(items, method, extra_data=[])
+    def json_for_autocomplete(items, method, extra_data=[], extra_methods=[])
       items = items.collect do |item|
         hash = HashWithIndifferentAccess.new({"id" => item.id.to_s, "label" => item.send(method), "value" => item.send(method)})
-        extra_data.each do |datum|
+        extra_methods.each do |datum|
```

Comment:
Well, as I stated above, `extra_data` is translated to columns names, but then in the code a resulting `Hash` is collected again by calling `methods`  from `extra_data`. Since we already have those columns, I decided to get rid of them and add custom `extra_methods`.